homeassistant/components/homekit/type_air_purifiers.py

```diff
@@ -214,202 +245,55 @@ def run(self) -> None:
 
         Run inside the Home Assistant event loop.
         """
-        if self.linked_humidity_sensor:
-            self._subscriptions.append(
-                async_track_state_change_event(
-                    self.hass,
-                    [self.linked_humidity_sensor],
-                    self._async_update_current_humidity_event,
-                    job_type=HassJobType.Callback,
-                )
-            )
-
-        if self.linked_pm25_sensor:
-            self._subscriptions.append(
-                async_track_state_change_event(
-                    self.hass,
-                    [self.linked_pm25_sensor],
-                    self._async_update_current_pm25_event,
-                    job_type=HassJobType.Callback,
-                )
-            )
-
-        if self.linked_temperature_sensor:
-            self._subscriptions.append(
-                async_track_state_change_event(
-                    self.hass,
-                    [self.linked_temperature_sensor],
-                    self._async_update_current_temperature_event,
-                    job_type=HassJobType.Callback,
+        # Add subscriptions for all float sensors
+        for linked_sensor, characteristic, post_processor in (
+            (self.linked_humidity_sensor, self.char_current_humidity, None),
+            (
+                self.linked_pm25_sensor,
+                self.char_pm25_density,
+                self._async_update_air_quality_from_pm25,
+            ),
+            (self.linked_temperature_sensor, self.char_current_temperature, None),
+            (
+                self.linked_filter_life_level_sensor,
+                self.char_filter_life_level,
+                self._async_update_filter_change_indicator_from_filter_life_level,
+            ),
+        ):
+            if linked_sensor and characteristic:
+                self.track_linked_entity(
+                    LinkedEntityStateTracking(
+                        linked_sensor, characteristic, convert_to_float, post_processor
+                    )
                 )
-            )
 
```

Comment:
I'd probably write this one out as it might be less code and the loop is a bit much to read.